packages/components/src/components/assets/AssetIcon/AssetIcon.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from 'react';
+import {
+    AssetShareIndicator,
+    AssetShareIndicatorProps,
+} from '../AssetShareIndicator/AssetShareIndicator';
+import { useTheme } from 'styled-components';
+
+export const UPDATED_ICONS_LIST_URL_BASE = 'https://data.trezor.io/suite/icons/coins/';
+
+const QUALITY = {
+    small: '',
+    big: '@2x',
+};
+
+const QUALITY_SIZE = {
+    small: 24,
+    big: 48,
+};
+
+type Quality = keyof typeof QUALITY;
+
+const ICON_FILETYPE = '.webp';
+
+type AssetIconProps = AssetShareIndicatorProps & {
+    coingeckoId?: string;
+    contractAddress?: string;
+    quality?: Quality;
+};
+
+const getAssetUrl = (coingeckoId: string, contractAddress: string, quality: Quality) => {
+    const fileName = contractAddress ? `${coingeckoId}_${contractAddress}` : coingeckoId;
+
+    return fileName
+        ? `${UPDATED_ICONS_LIST_URL_BASE}${fileName}${QUALITY[quality]}${ICON_FILETYPE}`
+        : '';
+};
+
+const useAvatarUrl = ({ coingeckoId, quality }: { coingeckoId?: string; quality: Quality }) => {
+    const theme = useTheme();
+
+    const defaultCharacter = '?';
+    const size = QUALITY_SIZE[quality];
+    const firstCharacter = coingeckoId ? coingeckoId[0] : defaultCharacter;
+    const fontColor = theme.iconSubdued.replace('#', '');
+    const backgroundColor = theme.backgroundSurfaceElevation0.replace('#', '');
+
+    const avatarUrl = `https://ui-avatars.com/api/?size=${encodeURIComponent(size)}&name=${encodeURIComponent(firstCharacter)}&bold=true&font-size=0.75&color=${encodeURIComponent(fontColor)}&background=${encodeURIComponent(backgroundColor)}&rounded=true`;
```

Comment:
I really don't like iinjecting external URLs into Suite.

Reply:
I agree with @peter-sanderson

Reply:
We might create component for it in DS if we are going to use it more than once.

Reply:
There are many more options to achieve the expected, it was just an idea while avoiding lots of css :)